### PR TITLE
Add carpets

### PR DESCRIPTION
**Pull request by lnjX**
Should them in the wool mod or default. I think it's better to have them in default.
They are as in MC.
This is TO-DO 1.

### CODE REVIEW

**Comment by LNJplus** [comment]
Tested: Works fine!

**Comment by BlockMen** [comment]
This should be added to the wool mod, not default. You can't rely wool_* tiles are always available.
Also you shouldn't add a second, local definition of used dyes.

**Comment by LNJplus** [comment]
I'm a git beginner and the commits aren't completely clear (I tried to work with git rebase), but the result should be fine.

**Comment by kilbith** [comment]
Too specific, bloats the inventory and impossibility to place stuff atop.

**Comment by BlockMen** [comment]
@kilbith as i said here, before decor (and/or many nodes) gets added the creative menu needs rework. https://github.com/BlockMen/minetest_next/issues/33#issuecomment-158467281

**Comment by BlockMen** [comment]
@LNJplus take a look at how to squash commits, that will help a lot ;)

Also, while you on doin that, please fix the codestyle of a few lines (will mark them).

File: mods/wool/init.lua [hunk 1]
```diff
@@ -48,3 +49,33 @@ for _, row in ipairs(wool.dyes) do
 	end
 end
 
+-- Register carpets
+for _, row in ipairs(wool.dyes) do
+	local name = row[1]
+	local desc = row[2]
+	
+	-- Node Definition
+	minetest.register_node("wool:carpet_"..name, {
```
Comment (BlockMen) [inline mods/wool/init.lua]: plz add blanks between ```"wool:carpet_"``` and ```..``` and at similar cases.

File: mods/wool/init.lua [hunk 2]
```diff
@@ -48,3 +49,33 @@ for _, row in ipairs(wool.dyes) do
...
+		description = desc.." Carpet",
+		tiles = {"wool_"..name..".png"},
```
Comment (BlockMen) [inline mods/wool/init.lua]: here aswell

File: mods/wool/init.lua [hunk 3]
```diff
@@ -48,3 +49,33 @@ for _, row in ipairs(wool.dyes) do
...
 end
 
+-- Register carpets
+for _, row in ipairs(wool.dyes) do
+	local name = row[1]
+	local desc = row[2]
+	
+	-- Node Definition
+	minetest.register_node("wool:carpet_"..name, {
+		description = desc.." Carpet",
```
Comment (BlockMen) [inline mods/wool/init.lua]: and here

File: mods/wool/init.lua [hunk 4]
```diff
@@ -48,3 +49,33 @@ for _, row in ipairs(wool.dyes) do
...
+		tiles = {"wool_"..name..".png"},
+		is_ground_content = true,
+		paramtype = "light",
+		drawtype = "nodebox",
+		node_box = {
+			type = "fixed",
+			fixed = {
+				{-0.5, -0.5, -0.5,  0.5, -0.57+2/16, 0.5},
```
Comment (BlockMen) [inline mods/wool/init.lua]: and here (around +)

**Comment by LNJplus** [comment]
Normally I do it with spaces, but not in this case (and some code is by c55). I hope I didn't forgot anything.

=============================

And a question should there be a comma (in the line with abc)?
```
bla = {
  {xyz},
  {abc},
}
```
If you write it so, definitely no:
```
bla = {{xyz}, {abc}}
```
I mostly write case one without comma, because you write case two without comma too. Is that right?

**Comment by BlockMen** [comment]
Hmm...i would decide this from case to case. If i expect that to that table (often) more content is added then i would add the comma in case one. (So most time i would add no comma)

In case two i would never add a comma, yes.

**Comment by LNJplus** [comment]
13 days nothing happened.
Is there something you disagree? :smiley_cat:

**Comment by BlockMen** [comment]
> I think the main focus should be on the improved creative inventory (12.) since adding more nodes (like the chairs, carpets, etc) would bloat it even more.

https://github.com/BlockMen/minetest_next/issues/33#issuecomment-158467281 ^

So it will go in when it does not bloat the inventory more. So either someone else does it or i do it myself when i have more time than currently.

**Pull request closed without merging**